focal: Add persistent GPU offloading

crawfxrd opened the pull request:
When `X-KDE-RunOnDiscreteGpu` is set in the application's .desktop file, launch the application on the discrete GPU if available, and provide the option to launch on the integrated GPU.

See also: #14

leviport commented:
Setting `X-KDE-RunOnDiscreteGpu=true` in `/usr/share/application/blender.desktop` does indeed make Blender launch on dGPU by default, but the launch option I see when right clicking still says "Launch using Dedicated Graphics Card". Clicking it does indeed launch it on the dGPU.

crawfxrd commented:
I rebased the patch (and fixed a typo). It wasn't applying cleanly after the last merge. It's working for me again.

leviport commented:
I'm getting the same behavior I was previously. Is there a cache I need to clear out or something? Should I try a fresh install?

crawfxrd commented:
Are you updating desktop entries after updating gnome-shell? If so, you need to start a new session (restart gnome-shell or logout/login). I don't know if there's a way to force re-reading the entries on the running session.

leviport commented:
I've rebooted and toggled graphics modes, but I'm still getting the same thing. I'm using the gaze14 with a 1660Ti for this, but that shouldn't be an issue, right?

crawfxrd committed:
Add persistent GPU offloading

When X-KDE-RunOnDiscreteGpu is set in the application's .desktop file,
launch the application on the discrete GPU if available, and provide the
option to launch on the integrated GPU.

crawfxrd commented:
_Actually_ fixed the typo this time (I forgot to add it to the patch file :no_mouth:).

leviport reviewed (approved):
Works great now. Awesome feature
